pkg/provisioningv2/rke2/planner/s3args.go

```diff
@@ -88,11 +88,11 @@ func getS3Credential(secretCache corecontrollers.SecretCache, namespace, name, r
 	}
 
 	secret, err := machineprovision.GetCloudCredentialSecret(secretCache, namespace, name)
-	if err != nil {
-		return result, fmt.Errorf("failed to lookup etcdSnapshotCloudCredentialName: %w", err)
-	}
-	if secret.Type != "provisioning.cattle.io/cloud-credential" && secret.Namespace != namespaces.GlobalNamespace {
-		return result, fmt.Errorf("expected secret of type provisioning.cattle.io/cloud-credential, got [%s]", secret.Type)
+	if err != nil || (secret.Type != "provisioning.cattle.io/cloud-credential" && secret.Namespace != namespaces.GlobalNamespace) {
```

Comment:
> Support is added for checking for a secret of the expected type or in the expected namespace.

You said "or" in the description but its using && here. Which is correct ? I'm confused by this conditional, couldn't the err exist and be a not found err but then you'd still check the secret.type ? 

Darren commented in other thread and it may not matter anymore.

Reply:
The secret needs to be of the expected type or in the expected namespace. So the negation of that is not the expected type and not in the expected namespace.

Also if the error is not nil, then the or will be true and we won't check the secret type.

Reply:
Now that I see Darren's comment then this doesn't matter. I will adjust it based on his comment.

Reply:
Ahh right, was reading it backwards.